google/cloud/spanner/instance.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INSTANCE_H_
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INSTANCE_H_
+
+#include "google/cloud/spanner/version.h"
+#include <ostream>
+#include <string>
+#include <tuple>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+/**
+ * This class identifies a Cloud Spanner Instance
+ *
+ * A Cloud Spanner instance is identified by its `project_id` and `instance_id`.
```

Comment:
Oh, this is good. Now it makes me think that the `Database` class should change to pair an `Instance` + a `database_id`. And `Database::ParentName` should change to an `Instance` getter. None of these changes are needed in this PR, but if you like this idea, we could file an issue to "fix" `Database` to take advantage of this new class more.

Reply:
Filed #586